blog(embertimes113): ember-backstop (#252)
ember-learn/ember-blog: source/2019-08-30-the-ember-times-issue-113.md

---

## [Ember-Backstop Visual Regression Testing Tutorial 🧪](https://www.linkedin.com/pulse/ember-backstop-visual-regression-testing-tutorial-garris-shipon/)

[@garris](https://github.com/garris) published a [blog post](https://www.linkedin.com/pulse/ember-backstop-visual-regression-testing-tutorial-garris-shipon/) on how to add visual regression tests to your Ember application in 10 minutes with the [ember-backstop](https://github.com/garris/ember-backstop) test-helper addon.

> Visual regression testing is done by importing `ember-backstop` into your application test and awaiting it at a point in time where we want to run the visual regression testing. It will then serialize the snapshot of your DOM at that moment in time and sending it to the *Backstop-Remote* service. The tutorial goes through how to verify, fix, and approve visual regression tests using the Backstop commands.

Reviewer: What do you think about

```suggestion
You can import `ember-backstop` into your application test and await it at a point in time where you want to run the visual regression testing. It will then serialize the snapshot of your DOM at that moment in time and it will send it to the *Backstop-Remote* service. The tutorial goes through how to verify, fix, and approve visual regression tests using the Backstop commands.
```
